src/test/java/mate/academy/onlinebookstore/service/book/BookServiceImplTest.java

```diff
@@ -0,0 +1,194 @@
+        requestDto.setTitle("New title");
+        requestDto.setPrice(BigDecimal.valueOf(25.00));
+        requestDto.setCategoryIds(List.of(1L, 2L));
+
+        Book book = new Book();
+        book.setTitle(requestDto.getTitle());
+        book.setAuthor(requestDto.getAuthor());
+        book.setIsbn(requestDto.getIsbn());
+        book.setPrice(requestDto.getPrice());
+        book.getCategories().add(category);
+
+        BookDto expected = initializeBookDto(book);
+        expected.setCategoryIds(List.of(1L, 2L));
+
+        when(bookRepository.findById(anyLong())).thenReturn(Optional.of(book));
+        when(categoryRepository.findById(anyLong())).thenReturn(Optional.of(category));
+        when(bookRepository.save(book)).thenReturn(book);
+        when(bookMapper.intoBookDto(book)).thenReturn(expected);
+
+        BookDto actual = bookService.updateBookById(requestDto, 1L);
+        assertThat(actual).isEqualTo(expected);
+        verify(bookRepository, times(1)).save(book);
+    }
+
+    @Test
+    @DisplayName("Get a list of books.")
+    void getAllBooks_ReturnsBookDtoList() {
+        final Book book1 = initializeBook();
+
+        final Book book2 = new Book();
+        book2.setTitle("Title 2");
+        book2.setAuthor("Author 2");
+        book2.setIsbn("ISBN 2");
+        book2.setPrice(BigDecimal.valueOf(25.00));
+        book2.getCategories().add(category);
+
+        BookDto bookDto1 = initializeBookDto(book1);
+
+        BookDto bookDto2 = initializeBookDto(book2);
+        bookDto2.setId(2L);
+
+        List<BookDto> expected = List.of(bookDto1, bookDto2);
+        Pageable pageable = PageRequest.of(0, 10);
+        Page<Book> bookPage = new PageImpl<>(List.of(book1, book2));
+
+        when(bookRepository.findAll(pageable)).thenReturn(bookPage);
+        when(bookMapper.intoBookDto(book1)).thenReturn(bookDto1);
+        when(bookMapper.intoBookDto(book2)).thenReturn(bookDto2);
+
+        List<BookDto> actual = bookService.findAllBooks(pageable);
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    @DisplayName("Verify saveBook method throws exception for invalid input.")
+    void saveBook_InvalidRequestDto_ThrowsException() {
+        CreateBookRequestDto requestDto = initializeCreateBookRequestDto();
+        requestDto.setTitle(null);
+
+        when(bookMapper.intoModel(requestDto)).thenThrow(IllegalArgumentException.class);
+
+        assertThrows(IllegalArgumentException.class, () -> bookService.saveBook(requestDto));
+        verify(bookRepository, never()).save(any());
+    }
+
+    @Test
+    @DisplayName("Verify saveBook method throws exception for non-existent category.")
+    void saveBook_NonExistentCategoryId_ThrowsEntityNotFoundException() {
+        CreateBookRequestDto requestDto = initializeCreateBookRequestDto();
+        requestDto.setCategoryIds(List.of(999L));
+
+        when(categoryRepository.findById(anyLong())).thenReturn(Optional.empty());
```

Comment:
it is better to avoid using methods like `any()`, `anyLong()` etc. Pass actual parameters instead. Fix everywhere
```suggestion
        Long invalidId = 999L;
        requestDto.setCategoryIds(List.of(invalidId));

        when(categoryRepository.findById(invalidId)).thenReturn(Optional.empty());
```